spec/models/collection_spec.rb

```diff
@@ -17,13 +19,28 @@
   describe 'attributes' do
     it do
       is_expected.to have_attributes(
+        id: 'agency_blogs',
         token: 'secret',
-        id: 'agency_blogs'
+        created_at: an_instance_of(Time),
+        updated_at: an_instance_of(Time)
       )
     end
   end
 
   describe 'validations' do
     it { is_expected.to validate_presence_of(:token) }
   end
+
+  describe '#last_document_sent' do
+    subject(:last_document_sent) { collection.last_document_sent }
+
+    context 'when something goes wrong' do
+      before do
+        allow_any_instance_of(DocumentRepository).
+          to receive(:search).and_raise(StandardError)
+      end
+
+      it { is_expected.to be nil }
```

Comment:
We lacked unit tests for `Collection#last_document_sent` and `Collection#document_total`. I added a missing spec for the failure case. I was tempted to refactor those methods into their own class and add additional unit tests, but held off as those methods are only used by a deprecated feature (I14y Drawer management from the admin center). (The methods are fully tested via our request specs.)